vendors/renesas/boards/rx65n-rsk/ports/ota_pal_for_aws/ota_pal.c

```diff
@@ -0,0 +1,1291 @@
+	uint32_t loop_counter = 0;
+	uint32_t index = 0;
+
+	/* search aligned FLASH_CF_MIN_PGM_SIZE top offset */
+	while(1)
+	{
+		if(!(tmp->content.offset % FLASH_CF_MIN_PGM_SIZE))
+		{
+			/* extract continuous flash_block candidate */
+			assembled_length = 0;
+			loop_counter = 0;
+			while(1)
+			{
+				if ((tmp != NULL) && (assembled_length < FLASH_CF_MIN_PGM_SIZE))
+				{
+					if(loop_counter < FLASH_CF_MIN_PGM_SIZE)
+					{
+						if((tmp->content.offset + tmp->content.length) > ((tmp->content.offset & OTA_FLASH_MIN_PGM_SIZE_MASK) + FLASH_CF_MIN_PGM_SIZE))
+						{
+							fragmented_length = (FLASH_CF_MIN_PGM_SIZE - assembled_length);
+							assembled_length += fragmented_length;
+							flash_block_candidate[loop_counter] = tmp;
+					        tmp->next = fragmented_flash_block_list_insert(tmp->next, tmp->content.offset + fragmented_length, &tmp->content.binary[fragmented_length], tmp->content.length - fragmented_length);
+							tmp->content.length = fragmented_length;
+						}
+						else
+						{
+							assembled_length += tmp->content.length;
+							flash_block_candidate[loop_counter] = tmp;
+						}
+					}
+					else
+					{
+						break;
+					}
+					tmp = tmp->next;
+					loop_counter++;
+				}
+				else
+				{
+					break;
+				}
+			}
+		}
+
+		/* break if found completed flash_block_candidate or found end of list */
+		if((assembled_length == FLASH_CF_MIN_PGM_SIZE) || (tmp == NULL))
+		{
+			break;
+		}
+		/* search next candidate */
+		else
+		{
+			tmp = tmp->next;
+		}
+	}
+
+	/* assemble flash_block */
+	if(assembled_length == FLASH_CF_MIN_PGM_SIZE)
+	{
+		tmp = head;
+		/* remove flash_block from list */
+		flash_block->offset = flash_block_candidate[0]->content.offset;
+		flash_block->length = assembled_length;
+		for(uint32_t i = 0; i < loop_counter; i++)
+		{
+			memcpy(&flash_block->binary[index], flash_block_candidate[i]->content.binary, flash_block_candidate[i]->content.length);
+			index = flash_block_candidate[i]->content.length;
+			tmp = fragmented_flash_block_list_delete(tmp, flash_block_candidate[i]->content.offset);
+		}
+	}
+	else
+	{
+		tmp = head;
+		flash_block->offset = NULL;
+		flash_block->binary = NULL;
+		flash_block->length = NULL;
+	}
+	return tmp;
+}
+
+static void ota_flashing_task( void * pvParameters )
+{
+	flash_err_t flash_err;
+	static uint8_t block[(1 << otaconfigLOG2_FILE_BLOCK_SIZE)];
+	static uint32_t ulOffset;
+	static uint32_t length;
+
+	while(1)
+	{
+		xQueueReceive(xQueue, &packet_block_for_queue2, portMAX_DELAY);
+		xSemaphoreTake(xSemaphoreFlashig, portMAX_DELAY);
+		memcpy(block, packet_block_for_queue2.p_packet, packet_block_for_queue2.length);
+		ulOffset = packet_block_for_queue2.ulOffset;
+		length = packet_block_for_queue2.length;
+		flash_err = R_FLASH_Write((uint32_t)block, ulOffset + BOOT_LOADER_UPDATE_TEMPORARY_AREA_LOW_ADDRESS + BOOT_LOADER_USER_FIRMWARE_HEADER_LENGTH, length);
+		if(packet_block_for_queue2.length != 1024)
+		{
+			nop();
```

Comment:
this seems like a bad error if the length does not match, but we do nothing and still add the length below? This seems weird? Looks like it only works happy path?

Reply:
Same response as the other comment I wrote.